### PR TITLE
[GH-1320] add snapshotReaders to tdr source

### PR DESCRIPTION
**Pull request by ehigham**
### Purpose
RR: https://broadinstitute.atlassian.net/browse/GH-1320
Adding a list of people who want to access to the snapshots we create to the source request.

### CODE REVIEW

**Review by rexwangcc — APPROVED** [review]

**Comment by rexwangcc** [comment]
Not sure why the other one gets closed, LGTM

File: api/src/wfl/module/covid.clj [hunk 1]
```diff
@@ -262,27 +263,28 @@
...
+  [{:keys [dataset table snapshotReaders] :as source} now-obj row-ids]
   (let [dt-format   (DateTimeFormatter/ofPattern "YYYYMMdd'T'HHmmss")
         compact-now (.format now-obj dt-format)]
     (letfn [(create-snapshot [idx shard]
               [shard (-> (format "_%s_%s" compact-now idx)
-                         (go-create-snapshot! dataset table shard))])]
+                         (go-create-snapshot! source shard))])]
       (->> row-ids
            (partition-all 500)
            (map vec)
```
Comment (tbl3rd) [inline api/src/wfl/module/covid.clj:290]: Is `snapshotReaders` used?

**Review by tbl3rd — APPROVED** [review]

File: api/src/wfl/module/covid.clj (hunk 1 above)
Comment (ehigham) [reply api/src/wfl/module/covid.clj:290]: Yes, in the request sent to TDR

File: api/src/wfl/module/covid.clj [hunk 2]
```diff
@@ -262,27 +263,28 @@
...
+   write job info as well as rows into `source-details-name` table.
+   Snapshots will be readable by members of the `snapshotReaders` list.
+   `suffix` will be appended to the snapshot names."
+  [suffix {:keys [dataset table snapshotReaders] :as _source} row-ids]
   (let [columns (->> (datarepo/all-columns dataset table)
                      (mapv :name)
                      (cons "datarepo_row_id"))]
     (-> (datarepo/make-snapshot-request dataset columns table row-ids)
         (update :name #(str % suffix))
+        (assoc :readers snapshotReaders)
```
Comment (ehigham) [inline api/src/wfl/module/covid.clj:276]: @tbl3rd

File: api/src/wfl/module/covid.clj (hunk 1 above)
Comment (tbl3rd) [reply api/src/wfl/module/covid.clj:290]: But not locally to `create-snapshots`.

File: api/src/wfl/module/covid.clj (hunk 2 above)
Comment (tbl3rd) [reply api/src/wfl/module/covid.clj:276]: But that's not in `create-snapshots` though this review UI obscures that.

**Pull request merged**